docs/Formatting.md

```diff
@@ -1,283 +1,335 @@
+
+  ```xml
+  <?xml version="1.0" encoding="UTF-8"?>
+  <root>
+    <child attribute='value' otherAttribute="otherValue"></child>
+  </root>
+  ```
+
+will be formatted to:
+
+  ```xml
+  <?xml version='1.0' encoding='UTF-8'?>
+  <root>
+    <child attribute='value' otherAttribute='otherValue'></child>
+  </root>
+  ```
+
+No changes to quotes will occur during formatting if `xml.format.enforceQuoteStyle` is set to `ignore`.
+
+***
+
+### xml.format.emptyElements
+
+Expand/collapse empty elements. Available values are `ignore`, `collapse` and `expand`. Defaults to `ignore`.
+An empty element is an element which is empty or which contains only white spaces.
+
+Set to `collapse` to collapse empty elements during formatting.
+
+  ```xml
+  <example attr="value" ></example>
+  ```
+  becomes...
+  ```xml
+    <example attr="value" />
+  ```
+
+Set to `expand` to expand empty elements during formatting.
+
+  ```xml
+  <example attr="value" />
+  ```
+  becomes...
+  ```xml
+   <example attr="value" ></example>
+  ```
+***
+
+### xml.format.preserveAttributeLineBreaks
+
+Preserve line breaks that appear before and after attributes. This setting is overridden if [xml.format.splitAttributes](Formatting#xmlformatsplitattributes-) is set to `true`. Default is `false`.
+
+If set to `true`, formatting does not change the following document:
+
+  ```xml
+  <?xml version='1.0' encoding='UTF-8'?>
+  <root>
+    <child
+      attr1='value1'
+      attr2='value2'
+      attr3='value3'></child>
+  </root>
+  ```
+
+If set to `false`, the document above becomes:
+
+  ```xml
+  <?xml version='1.0' encoding='UTF-8'?>
+  <root>
+    <child attr1='value1' attr2='value2' attr3='value3'></child>
+  </root>
+  ```
+
+***
+
+### xml.format.preservedNewlines
+
+The number of blank lines to leave between tags during formatting.
+
+The default is 2. This means that if more than two consecutive empty lines are left in a document, then the number of blank lines will become 2.
+
+  For example, this document:
+  ```xml
+  <?xml version="1.0" encoding="UTF-8"?>
+
+
+
+  <root>
+
+
+
+    <child></child>
+
+
+
+  </root>
+  ```
+
+  Will be replaced with:
+  ```xml
+  <?xml version="1.0" encoding="UTF-8"?>
+
+
+  <root>
+
+
+    <child></child>
+
+
+  </root>
+  ```
+
+If this value is set to 0, then all blank lines will be removed during formatting.
+
+  For example, this document:
+  ```xml
+  <?xml version="1.0" encoding="UTF-8"?>
+
+  <root>
+
+    <child></child>
+
+  </root>
+  ```
+
+  Will be replaced with:
+  ```xml
+  <?xml version="1.0" encoding="UTF-8"?>
+  <root>
+    <child></child>
+  </root>
+  ```
+
+***
+
+### xml.format.splitAttributes
+
+  Set to `true` to split node attributes onto multiple lines during formatting. Defaults to `false`.
+  Overrides the behaviour of [xml.format.preserveAttributeLineBreaks](Formatting#xmlformatpreserveattributelinebreaks)
```

Comment:
The link doesn't work.

Reply:
Is there a mechanism to link between docs in the embedded documentation? I noticed something similar in `Preferences.md`

Reply:
For now I just removed the links in `Preferences.md` (for just the embedded docs). I should probably figure out how to replace the markdown content at extension run time so that links instead trigger a function that switches the content of the webview.